### PR TITLE
GCW-2781 Add & sync status column to package_types table & use in stock app

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-2781

### What does this PR do?

FEATURE: Sync StockIt's `codes.status` field with a new field in GoodCity `package_types.allow_stock`
Also added rake task to update `package_types.allow_stock` with Stockit `codes.status`.

### Impacted Areas
PackageType selection during in stock app will have types of `allow_stock` True.

### Linked PR's:
https://bitbucket.org/crfdevs/stockit/pull-requests/56/gcw-2781-add-sync-status-column-to/diff

Please review this PR.

### CODE REVIEW

File: lib/tasks/package_types.rake [hunk 1]
```diff
@@ -27,4 +27,27 @@ namespace :goodcity do
...
+  desc 'Update PackageType `allow_stock` according to `Codes.status` from Stockit'
+  task update_allow_stock_to_stockit_codes_status: :environment do
+    STATUS_AND_ALLOW_STOCK_MAPPING = {
+      "Active" => true,
+      "Inactive" => false
+    }
+    codes_json = Stockit::CodeSync.index
+    stockit_codes = JSON.parse(codes_json["codes"]) || {}
+    stockit_codes.each do |code|
+      if active?(code["status"])
```
Comment (swatijadhav) [inline lib/tasks/package_types.rake]: no need to have a method call for this.. 
```suggestion
      if STATUS_AND_ALLOW_STOCK_MAPPING[code["status"]]
```

Comment (patrixr) [reply lib/tasks/package_types.rake]: `active?(...)` is much more readable and understandable for me. I think having a mapping for what is essentially a boolean value is overkill

**Comment by patrixr** [comment]
@abulsayyad123 I kinda find the `allow_stock` property name a bit confusing.

Would something like `enabled` maybe be more appropriate ? 🤔 
Just thinking out loud.

**Comment by abulsayyad123** [comment]
@patrixr , `allow_stock` refers to allowing it for stock app.

File: lib/tasks/package_types.rake (hunk 1 above)
Comment (abulsayyad123) [reply lib/tasks/package_types.rake]: I added `active?` method to make its intention more obvious. I think its fine to with this method.

**Comment by abulsayyad123** [comment]
Closing this PR and creating the new PR for the same.
New PR: https://github.com/crossroads/api.goodcity/pull/836
CC: @namrataukirde

**Pull request closed without merging**